provider/internal/java/dependency.go

```diff
@@ -97,52 +96,67 @@ func getMavenLocalRepoPath(mvnSettingsFile string) string {
 	return string(outb.String())
 }
 
-func (p *javaServiceClient) GetDependencyFallback(ctx context.Context) (map[uri.URI][]*provider.Dep, error) {
-	pomDependencyQuery := "//dependencies/dependency/*"
-	path := p.findPom()
-	file := uri.File(path)
+func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, location string) (map[uri.URI][]*provider.Dep, error) {
+	deps := []*provider.Dep{}
 
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return nil, err
+	path := p.findPom()
+	if location != "" {
+		path = location
 	}
-
-	f, err := os.Open(absPath)
+	pom, err := gopom.Parse(path)
 	if err != nil {
 		return nil, err
 	}
-	doc, err := xmlquery.Parse(f)
-	if err != nil {
-		return nil, err
+
+	// have to get both <dependencies> and <dependencyManagement> dependencies (if present)
+	var pomDeps []gopom.Dependency
+	if pom.Dependencies != nil {
+		pomDeps = append(pomDeps, *pom.Dependencies...)
 	}
-	list, err := xmlquery.QueryAll(doc, pomDependencyQuery)
-	if err != nil {
-		return nil, err
+	if pom.DependencyManagement != nil {
+		if pom.DependencyManagement.Dependencies != nil {
+			pomDeps = append(pomDeps, *pom.DependencyManagement.Dependencies...)
+		}
 	}
-	deps := []*provider.Dep{}
-	dep := &provider.Dep{}
-	// TODO this is comedically janky
-	for _, node := range list {
-		if node.Data == "groupId" {
-			if dep.Name != "" {
-				deps = append(deps, dep)
-				dep = &provider.Dep{}
+
+	// add each dependency found
+	for _, d := range pomDeps {
+		dep := provider.Dep{}
+		dep.Name = fmt.Sprintf("%s.%s", *d.GroupID, *d.ArtifactID)
+		if *d.Version != "" {
+			if strings.Contains(*d.Version, "$") {
+				version := strings.TrimSuffix(strings.TrimPrefix(*d.Version, "${"), "}")
+				version = pom.Properties.Entries[version]
+				if version != "" {
+					dep.Version = version
+				}
+			} else {
+				dep.Version = *d.Version
 			}
-			dep.Name = node.InnerText()
-		} else if node.Data == "artifactId" {
-			dep.Name += "." + node.InnerText()
-		} else if node.Data == "version" {
-			dep.Version = node.InnerText()
 		}
-		// Ignore the others
-	}
-	if !reflect.DeepEqual(dep, provider.Dep{}) {
-		dep.Labels = []string{fmt.Sprintf("%v=%v", provider.DepSourceLabel, javaDepSourceInternal)}
-		deps = append(deps, dep)
+		deps = append(deps, &dep)
 	}
+
 	m := map[uri.URI][]*provider.Dep{}
-	m[file] = deps
+	m[uri.File(path)] = deps
```

Comment:
I am a little worried about this change having side effects. is there a reason to move to a URI instead of the file path that is passed in?

Reply:
This is the same, I just removed `file := uri.File(path)` [above](https://github.com/konveyor/analyzer-lsp/pull/368/files#diff-b70ad3b19e5302fa83ea4cbeae98cbd7bb689d2d888102229fe3f133f771f44eL103).